README.md

```diff
@@ -1,9 +1,232 @@
 
-libfabric
-=========
+# libfabric
 
-The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric communication services to applications.
+The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric
+communication services to applications.
 
-See [the OFI web site](http://ofiwg.github.io/libfabric/) for more details, including a description and overview of the project, and detailed documentation of the libfabric APIs.
+See [the OFI web site](http://libfabric.org) for more details, including a
+description and overview of the project, and detailed documentation of the
+Libfabric APIs.
+
+## Installing libfabric
+
+On OS X the latest release of Libfabric can be installed using the
+[Homebrew](https://github.com/Homebrew/homebrew) package manager using the
+following command:
+
+```
+brew install libfabric
+```
+
+## Building libfabric
+
+If you are building Libfabric from a developer Git clone, you must first run
+the `autogen.sh` script. This will invoke the GNU Autotools to bootstrap
+Libfabric's configuration and build mechanisms. If you are building Libfabric
+from an official distribution tarball from libfabric.org, there is no need to
+run `autogen.sh`; Libfabric distribution tarballs are already bootstrapped for
+you.
+
+Libfabric currently supports GNU/Linux, Free BSD, and OS X.
+
+### Dependencies
+
+- `libpthread`
+
+Provider dependencies will be discussed in the the "Providers" section.
+
+### Configure options
+
+The `configure` script has many built in options (see `./configure --help`).
+Some useful options are:
+
+```
+--prefix=<directory>
+```
+
+By default `make install` will place the files in the `/usr` tree.
+The `--prefix` option specifies that Libfabric files should be installed into
+the tree specified by named `<directory>`. The executables will be located at
+`<directory>/bin`.
+
+```
+--with-valgrind=<directory>
+```
+
+Directory where valgrind is installed. If valgrind is found, then valgrind
+annotations are enabled. This may incur a performance penalty.
+
+```
+--enable-debug
+```
+
+Enable debug code paths. This enables various extra checks and allows for using
+the highest verbosity logging output that is normally compiled out in
+production builds.
+
+```
+--enable-<provider>=[yes|no|auto|dl|<directory>]
+--disable-<provider>
+```
+
+This enables or disables the provider named `<provider>`. Valid options are:
+- Auto (This is the default if the `--enable-<provider>` option isn't specified)
+
+  The provider will be enabled if all of its requirements are satisfied. If one
+  of the requirements cannot be satisfied, then the provider is disabled.
+- Yes (This is the default if the `--enable-<provider>` option is specified)
+
+  The configure script will abort if the provider cannot be enabled (e.g., due
+  to some of its requirements not being available.
+- No
+
+  Disable the provider.
```

Comment:
You should mention that `--disable-<provider>` is a synonym for `--enable-<provider>=no`